kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/targetDefaultForClassName.kt

```diff
@@ -51,11 +50,10 @@ fun targetDefaultForType(providedArbs: Map<KClass<*>, Arb<*>> = emptyMap(), type
       type == typeOf<Period>() -> Arb.period()
       type == typeOf<BigDecimal>() -> Arb.bigDecimal()
       type == typeOf<BigInteger>() -> Arb.bigInt(maxNumBits = 256)
-      clazz?.isData == true -> {
```

Comment:
If type matches are split off and handled first, maybe clazz can be handled with a single null check?